src/main/java/nwoolcan/model/brewery/production/batch/review/BatchEvaluationBuilder.java

```diff
@@ -0,0 +1,154 @@
+package nwoolcan.model.brewery.production.batch.review;
+
+import nwoolcan.utils.Result;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+/**
+ * {@link BatchEvaluation} builder.
+ * Handles all error checking for the construction of a new {@link BatchEvaluation}.
+ */
+
+public final class BatchEvaluationBuilder {
+    private static final String INVALID_SCORE_MESSAGE = "Score for one or more categories is not valid";
+    private static final String INVALID_CATEGORIES_MESSAGE = "Invalid categories";
+    private static final String BUILDER_USED_MESSAGE = "This builder has already built";
+    private static final BatchEvaluationScanner SCANNER = new BatchEvaluationScannerImpl();
```

Comment:
consider a singleton for this functionality

Reply:
a singleton for what? Scanner?

Reply:
if you need it i mean
still can be just a class not an interface if it has only one implementation an one use point

Reply:
> Bisogna esser sicuri che lo sia, però, e nel dubbio fra usare singleton e non usarlo è meglio non usarlo. 

cit D. Pianini and M. Fiordarancio
Two `BatchEvaluationScanner` instances won't be a problem and if used only in `BatchEvaluationBuilder` the static field prevent it from being initialized every time you create a builder